Sources/PackageModel/Manifest.swift

```diff
@@ -164,6 +206,54 @@ extension Manifest {
         try container.encode(platforms, forKey: .platforms)
     }
 
+    /// Returns the targets required for building the provided products.
+    public func targetsRequired(for products: [ProductDescription]) -> [TargetDescription] {
+        let targetsByName = Dictionary(targets.map({ ($0.name, $0) }), uniquingKeysWith: { $1 })
+        let productTargetNames = products.flatMap({ $0.targets })
+
+        let dependentTargetNames = transitiveClosure(productTargetNames, successors: { targetName in
+            targetsByName[targetName]?.dependencies.compactMap({ dependency in
+                switch dependency {
+                case .target(let name),
+                     .byName(let name):
+                    if targetsByName[name] != nil {
+                        return name
+                    } else {
+                        return nil
+                    }
```

Comment:
You can  also do `targetsByName.keys.contains(name) ? name : nil`

Reply:
Correct! That's better.